From 111339b292b3060c13c0e0bfeaddbba6e8de1dd1 Mon Sep 17 00:00:00 2001
From: Michael Sarahan <msarahan@gmail.com>
Date: Sat, 10 Aug 2019 13:37:43 -0500
Subject: [PATCH] make directory arg optional

---
 constructor/main.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/constructor/main.py b/constructor/main.py
index 01258e1..a08b3ac 100644
--- a/constructor/main.py
+++ b/constructor/main.py
@@ -198,6 +198,8 @@ def main():
     p.add_argument('dir_path',
                    help="directory containing construct.yaml",
                    action="store",
+                   nargs="?",
+                   default=os.getcwd(),
                    metavar='DIRECTORY')
 
     args = p.parse_args()
-- 
2.20.1

