diff --git a/constructor/header.sh b/constructor/header.sh
index 22b8bc4..2b98827 100644
--- a/constructor/header.sh
+++ b/constructor/header.sh
@@ -424,6 +424,20 @@ install_dist()
             exit 1
         fi
     fi
+    # FIXME: Temporarily move site-packages to lib/python-VER/site-packages to
+    # make noarch:python packages work. But this shall be fixed by upstream by
+    # either supporting noarch:python packages or blacklisting noarch:python
+    # packages.
+    if [ -z "${PYTHON_VERSION}" -a -f "${PYTHON}" ]; then
+      PYTHON_VERSION=$("${PYTHON}" -c 'import sys;v=sys.version_info;print("%d.%d"%v[:2])')
+    fi
+    if [ -d "${PREFIX}/site-packages" ]; then
+      pushd ${PREFIX} >/dev/null
+      mkdir -p "${PREFIX}/lib/python${PYTHON_VERSION}"
+      tar -c site-packages | tar -xf - -C "${PREFIX}/lib/python${PYTHON_VERSION}" || exit 1
+      rm -rf site-packages || exit 1
+      popd >/dev/null
+    fi
 #if not keep_pkgs
     rm "$PKG"
 #endif
