From b74f22a336ef9a865e37d96da84d1727f2512df2 Mon Sep 17 00:00:00 2001
From: Isuru Fernando <isuruf@gmail.com>
Date: Sat, 22 Oct 2022 18:36:23 -0500
Subject: [PATCH] set modification time for cache file for mamba

---
 constructor/conda_interface.py | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/constructor/conda_interface.py b/constructor/conda_interface.py
index 66a442a3..3b712888 100644
--- a/constructor/conda_interface.py
+++ b/constructor/conda_interface.py
@@ -7,6 +7,7 @@
 from copy import deepcopy
 from itertools import chain
 from os.path import join
+import datetime
 
 from constructor.utils import hash_files
 
@@ -134,17 +135,24 @@ def write_repodata(cache_dir, url, full_repodata, used_packages, info):
         url = used_repodata.pop('_url').rstrip("/")
         used_repodata.pop("_mod", None)
         repodata = json.dumps(used_repodata, indent=2)
+        mod_time = "Mon, 07 Jan 2019 15:22:15 GMT"
         repodata_header = json.dumps(
             {
-                "_mod": "Mon, 07 Jan 2019 15:22:15 GMT",
+                "_mod": mod_time,
                 "_url": url,
             }
         )
         repodata = repodata_header[:-1] + "," + repodata[1:]
-        repodata_filename = _cache_fn_url(url)
-        with open(join(cache_dir, repodata_filename), 'w') as fh:
+        repodata_filepath = join(cache_dir, _cache_fn_url(url))
+        with open(repodata_filepath, 'w') as fh:
             fh.write(repodata)
 
+        # set the modification time to mod_time. needed for mamba
+        mod_time_datetime = datetime.datetime.strptime(mod_time,
+            "%a, %d %b %Y %H:%M:%S %Z")
+        mod_time_s = int(mod_time_datetime.timestamp())
+        os.utime(repodata_filepath, times=(mod_time_s, mod_time_s))
+
     def write_cache_dir():
         cache_dir = join(PackageCacheData.first_writable().pkgs_dir, 'cache')
         mkdir_p_sudo_safe(cache_dir)
