From 1e0273366ed0f8e9836e5aa491aa12030b379c91 Mon Sep 17 00:00:00 2001
From: Isuru Fernando <isuruf@gmail.com>
Date: Sat, 8 Feb 2020 02:55:25 -0600
Subject: [PATCH] Fix aarch64 selector

---
 constructor/construct.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/constructor/construct.py b/constructor/construct.py
index f0f5f99..0213510 100644
--- a/constructor/construct.py
+++ b/constructor/construct.py
@@ -218,7 +218,7 @@ def ns_platform(platform):
         linux32 = bool(p == 'linux-32' or p == 'linux-armv7l'),
         linux64 = bool(p == 'linux-64' or p == 'linux-ppc64le'),
         armv7l = bool(p == 'linux-armv7l'),
-        aarch64 = bool(p == 'aarch64'),
+        aarch64 = bool(p == 'linux-aarch64'),
         ppc64le = bool(p == 'linux-ppc64le'),
         x86 = p.endswith(('-32', '-64')),
         x86_64 = p.endswith('-64'),
