From c1446149c506adcc6eeabc0fee1b96a5db22b36a Mon Sep 17 00:00:00 2001
From: Marcel Bargull <marcel.bargull@udo.edu>
Date: Fri, 3 Nov 2017 15:41:40 +0100
Subject: [PATCH] remove broken test test_link_idists

---
 constructor/tests/test_install.py | 3 ---
 1 file changed, 3 deletions(-)

diff --git constructor/tests/test_install.py constructor/tests/test_install.py
index 3023ea1..b41c580 100644
--- constructor/tests/test_install.py
+++ constructor/tests/test_install.py
@@ -129,9 +129,6 @@ def test_name_dist(self):
         self.assertEqual(name_dist('conda-build-1.21.6-py35_0'),
                          'conda-build')
 
-    def test_link_idists(self):
-        self.assertRaises(NotImplementedError, link_idists)
-
 
 def run():
     suite = unittest.TestSuite()
